src/utils/daos_hdlr.c

```diff
@@ -1869,46 +1872,416 @@ set_dm_args_default(struct dm_args *dm)
+ * the user has permissions.
+ */
+int
+dm_cont_get_all_props(struct cmd_args_s *ap, daos_handle_t coh, daos_prop_t **_props,
+		      bool get_oid, bool get_label, bool get_roots)
+{
+	int		rc;
+	daos_prop_t	*props = NULL;
+	daos_prop_t	*prop_acl = NULL;
+	daos_prop_t	*props_merged = NULL;
+	uint32_t        total_props = NUM_SERIALIZE_PROPS;
+	/* minimum number of properties that are always allocated/used to start count */
+	int             prop_index = NUM_SERIALIZE_PROPS;
+
+	if (get_oid)
+		total_props++;
+
+	/* container label is required to be unique, so do not retrieve it for copies.
+	 * The label is retrieved for serialization, but only deserialized if the label
+	 * no longer exists in the pool
+	 */
+	if (get_label)
+		total_props++;
+
+	if (get_roots)
+		total_props++;
+
+	/* Allocate space for all props except ACL. */
+	props = daos_prop_alloc(total_props);
+	if (props == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
+
+	/* The order of properties MUST match the order expected by serialization  */
+	props->dpp_entries[0].dpe_type = DAOS_PROP_CO_EC_CELL_SZ;
+	props->dpp_entries[1].dpe_type = DAOS_PROP_CO_LAYOUT_TYPE;
+	props->dpp_entries[2].dpe_type = DAOS_PROP_CO_LAYOUT_VER;
+	props->dpp_entries[3].dpe_type = DAOS_PROP_CO_CSUM;
+	props->dpp_entries[4].dpe_type = DAOS_PROP_CO_CSUM_CHUNK_SIZE;
+	props->dpp_entries[5].dpe_type = DAOS_PROP_CO_CSUM_SERVER_VERIFY;
+	props->dpp_entries[6].dpe_type = DAOS_PROP_CO_REDUN_FAC;
+	props->dpp_entries[7].dpe_type = DAOS_PROP_CO_REDUN_LVL;
+	props->dpp_entries[8].dpe_type = DAOS_PROP_CO_SNAPSHOT_MAX;
+	props->dpp_entries[9].dpe_type = DAOS_PROP_CO_COMPRESS;
+	props->dpp_entries[10].dpe_type = DAOS_PROP_CO_ENCRYPT;
+	props->dpp_entries[11].dpe_type = DAOS_PROP_CO_OWNER;
+	props->dpp_entries[12].dpe_type = DAOS_PROP_CO_OWNER_GROUP;
+	props->dpp_entries[13].dpe_type = DAOS_PROP_CO_DEDUP;
+	props->dpp_entries[14].dpe_type = DAOS_PROP_CO_DEDUP_THRESHOLD;
+
+	/* Conditionally get the OID. Should always be true for serialization. */
+	if (get_oid) {
+		props->dpp_entries[prop_index].dpe_type = DAOS_PROP_CO_ALLOCED_OID;
+		prop_index++;
+	}
+
+	if (get_label) {
+		props->dpp_entries[prop_index].dpe_type = DAOS_PROP_CO_LABEL;
+		prop_index++;
+	}
+
+	if (get_roots) {
+		props->dpp_entries[prop_index].dpe_type = DAOS_PROP_CO_ROOTS;
+	}
+
+	/* Get all props except ACL first. */
+	rc = daos_cont_query(coh, NULL, props, NULL);
+	if (rc != 0) {
+		DH_PERROR_DER(ap, rc, "Failed to query container");
+		D_GOTO(out, rc);
+	}
+
+	/* Fetch the ACL separately in case user doesn't have access */
+	rc = daos_cont_get_acl(coh, &prop_acl, NULL);
+	if (rc == 0) {
+		/* ACL will be appended to the end */
+		props_merged = daos_prop_merge(props, prop_acl);
+		if (props_merged == NULL) {
+			rc = -DER_INVAL;
+			DH_PERROR_DER(ap, rc, "Failed set container ACL");
+			D_GOTO(out, rc);
+		}
+		daos_prop_free(props);
+		props = props_merged;
+	} else if (rc && rc != -DER_NO_PERM) {
```

Comment:
Just trying to understand the logic here, wouldn't this be the same and easier to read?
```suggestion
	} else if (rc != -DER_NO_PERM) {
```